slides-2022/intro.md

```diff
@@ -0,0 +1,199 @@
+
+You will need to write at least 1000 lines of code, **by yourself** to complete
+homework. If this is not what you signed up for, there is still a option to
+de-register.
+
+---
+class: about-us
+# About us
+
+.cols[
+.fifty[
+### Artem Halas
+- Software engineer in Pipedrive
+- Love cycling
+
+.full-image[![Artem Halas](assets/artem_halas.png)]
+]
+
+.fifty[
+### Stanislav Deviatykh
+- Software engineer in Qminder
+- Love pixel art
+
+.full-image[![Stanislav Deviatykh](assets/stanislav_deviatykh.jpg)]
+]
+]
+
+---
+
+# About Pipedrive
+- First CRM platform made for Salespeople
+- Estonian #5 Unicorn
+- 10 Offices in 8 countries
+- 900 + employees
+- Used by 95 000 Companies
+
+--
+# About Qminder
+- B2B SaaS queue management system
+- 100+ clients from 50+ countries
+- 80+ million visitors served
+
+---
+
+# Course overview
+
+* JavaScript crash course
+  * Basic syntax
+  * Ecosystem & tooling
+* UI components using React
+* Means of composition
+* Testing, testing, testing
+--
+
+* Application state management
+* Cross-cutting concerns
+* Integration with external APIs
+* Asynchronous code
+
+The latter parts of the course are subject to change
+
+---
+
+# Course overview
+
+* Working from principles over hacking things together
+* A lot of (unit) testing which will be graded
+* Functional programming
+* Composition of simple ideas
+
+---
+
+# Course overview
+
+* Discussed topics are not specific to the browser nor JavaScript
+  * Many functional programming patterns are similar
+* For better or for worse, the Web is the most popular and accessible platform
+* This necessitates learning some JavaScript and the DOM (Document Object Model)
+
+---
+
+# Calendar
+
+A lot of work!
+
+* 3 EAP = 78 hours
+* 1 homework ~every week
+
+---
+
+# Homework
+
+* 7 homework, 10 points each
+* Continuous software project
+* First one given out today
+* ~8 hours spent on each homework
+* Hard deadline 23:59 the next Sunday (10 days)
+  * Miss deadline = 10 points gone from final grade
+* Catch-up application published Monday morning
+
+???
+
+The catch-up application is the only piece of code that can be used without
+attribution. It is there to avoid people who fail to produce a good enough
+solution one week from failing the whole course.
+
+---
+
+# Homework
+
+* No hand holding!
+* Slack workspace #help-homework for support
+* A lot of reading and figuring things out from zero
+* Straight to cold water!
+* Practical project lead experience
+
+---
+
+## Tips
+
+* Start early, homework simply cannot be accepted after deadline
+* Code style matters!
+  * This includes putting effort variable naming
+* What cannot be understood, cannot be graded
+
+---
+
+# Homework
+
+* We've been reading and reviewing code for a long time
+* We will be reading and reviewing homework from each one of you
+* It is easy for me to detect when code was pieced together from multiple sources
+* I've had to deal with **plagiarism** **every year**
```

Comment:
What do you think about changing to "We've had to deal" or maybe removing this line?